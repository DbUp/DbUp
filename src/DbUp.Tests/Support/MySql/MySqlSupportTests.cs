using System;
using ApprovalTests;
using DbUp.Tests.TestInfrastructure;
using NUnit.Framework;
using Shouldly;

namespace DbUp.Tests.Support.MySql
{
    [TestFixture]
    public class MySqlSupportTests
    {
        [Test]
        public void CanHandleDelimiter()
        {
            var recordingDbConnection = new RecordingDbConnection(true);
            var upgrader = DeployChanges.To
                .MySqlDatabase(string.Empty)
                .OverrideConnectionFactory(recordingDbConnection)
                .WithScript("Script0003", @"USE `test`;
DROP procedure IF EXISTS `testSproc`;

DELIMITER $$

USE `test`$$
CREATE PROCEDURE `testSproc`(
        IN   ssn                    VARCHAR(32)
     )
BEGIN 

    SELECT id      
    FROM   customer as c
    WHERE  c.ssn = ssn ; 

END$$").Build();

            var result = upgrader.PerformUpgrade();

            result.Successful.ShouldBe(true);
            var commandLog = recordingDbConnection.GetCommandLog();
            try
            {
                Approvals.Verify(commandLog, Scrubbers.ScrubDates);
            }
            catch (Exception)
            {
                Console.WriteLine(commandLog);
                throw;
            }
        }
    }
}
