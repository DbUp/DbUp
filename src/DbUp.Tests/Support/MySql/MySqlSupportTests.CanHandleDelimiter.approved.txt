Open connection
Execute scalar command: select 1 from INFORMATION_SCHEMA.TABLES where TABLE_NAME = 'schemaversions'
Dispose command
Execute non query command: USE `test`
Dispose command
Execute non query command: DROP procedure IF EXISTS `testSproc`
Dispose command
Execute non query command: USE `test`
Dispose command
Execute non query command: CREATE PROCEDURE `testSproc`(
        IN   ssn                    VARCHAR(32)
     )
BEGIN 

    SELECT id      
    FROM   customer as c
    WHERE  c.ssn = ssn ; 

END
Dispose command
Execute scalar command: select 1 from INFORMATION_SCHEMA.TABLES where TABLE_NAME = 'schemaversions'
Dispose command
Execute non query command: CREATE TABLE `schemaversions` 
                    (
                        `schemaversionid` INT NOT NULL AUTO_INCREMENT,
                        `scriptname` VARCHAR(255) NOT NULL,
                        `applied` TIMESTAMP NOT NULL,
                        PRIMARY KEY (`schemaversionid`));
Dispose command
Create parameter
Add parameter to command: scriptName=Script0003
Create parameter
Add parameter to command: applied=<date>
Execute non query command: insert into `schemaversions` (ScriptName, Applied) values (@scriptName, @applied)
Dispose command
Dispose connection